crates/red_knot/tests/file_watching.rs

```diff
@@ -42,19 +43,60 @@ impl TestCase {
     }
 
     fn stop_watch(&mut self) -> Vec<watch::ChangeEvent> {
-        if let Some(watcher) = self.watcher.take() {
-            // Give the watcher some time to catch up.
-            std::thread::sleep(Duration::from_millis(10));
-            watcher.flush();
-            watcher.stop();
+        self.try_stop_watch(Duration::from_secs(10))
+            .expect("Expected watch changes but observed none.")
+    }
+
+    fn try_stop_watch(&mut self, timeout: Duration) -> Option<Vec<watch::ChangeEvent>> {
+        let watcher = self.watcher.take().unwrap();
```

Comment:
I'll add a comment. Althought his isn't new code. 

The reason why it is an `Option` is because `watcher.stop` consumes the watcher and I didn't want to make the method take a `self` because it would consume the `TestCase` (which deletes the tempdir). 

The invariant here is that you can only call `stop_watch` once. Calling it a second time panics because the watcher isn't running anymore.

Reply:
> Althought his isn't new code.

Well, the `.unwrap()` call you're introducing in this PR is new. Previously you used `if let Some(watcher) = self.watcher.take()`, which implied it was okay to call `stop_watch()` several times — it just wouldn't have have done anything the second time, I suppose.

> The invariant here is that you can only call `stop_watch` once. Calling it a second time panics because the watcher isn't running anymore.

Great, thanks. Maybe we could use `.expect("Cannot call stop_watch() more than once!")` instead of `.unwrap()`?